libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -784,10 +788,8 @@ bool ContractCompiler::visit(Return const& _return)
 		for (auto const& retVariable: boost::adaptors::reverse(returnParameters))
 			CompilerUtils(m_context).moveToStackVariable(*retVariable);
 	}
-	for (unsigned i = 0; i < m_stackCleanupForReturn; ++i)
-		m_context << Instruction::POP;
-	m_context.appendJumpTo(m_returnTags.back());
-	m_context.adjustStackOffset(m_stackCleanupForReturn);
+
+	popAndJump(m_context.stackHeight() - m_returnTags.back().second, m_returnTags.back().first);
```

Comment:
It might be less error prone to only supply `m_returnTags.back().second` to `popAndJump` and do `m_context.stackHeight() - ...` inside `popAndJump`.